storage/rocksdb/ha_rocksdb.cc

```diff
@@ -204,51 +205,81 @@ static std::string rdb_normalize_dir(std::string dir) {
   return dir;
 }
 
-static int
-rocksdb_create_checkpoint(THD *const thd MY_ATTRIBUTE((__unused__)),
-                          struct SYS_VAR *const var MY_ATTRIBUTE((__unused__)),
-                          void *const save MY_ATTRIBUTE((__unused__)),
-                          struct st_mysql_value *const value) {
+static int rocksdb_create_checkpoint(const char *checkpoint_dir_raw) {
+  DBUG_ASSERT(checkpoint_dir_raw);
+
+  const auto checkpoint_dir = rdb_normalize_dir(checkpoint_dir_raw);
+  LogPluginErrMsg(INFORMATION_LEVEL, 0,
+                  "creating checkpoint in directory: %s\n",
+                  checkpoint_dir.c_str());
+  rocksdb::Checkpoint *checkpoint;
+  auto status = rocksdb::Checkpoint::Create(rdb, &checkpoint);
+  // We can only return HA_EXIT_FAILURE/HA_EXIT_SUCCESS here which is why
+  // the return code is ignored, but by calling into rdb_error_to_mysql,
+  // it will call my_error for us, which will propogate up to the client.
+  int rc;
+  (void)rc;
```

Comment:
no no no :)
My comment was to remove rc variable altogether and cast to void its assignments below

Reply:
`(void)some_function(asdf);` will still raise the `error: ignoring return value of` error due to the prototype explicitly containing `MY_ATTRIBUTE((__warn_unused_result__))` which overrides the void cast.  See interesting discussion here https://gcc.gnu.org/bugzilla/show_bug.cgi?id=25509

It could be argued in this case that the `MY_ATTRIBUTE((__warn_unused_result__))` is not necessarily correct for this method or, probably more correct, is that we shouldn't be using this method here and instead just directly calling `my_error(ER_GET_ERRMSG, MYF(0), status.code(), status.ToString().c_str(), rocksdb_hton_name);` which is all that `ha_rocksdb::rdb_error_to_mysql` is effectively doing here.

Reply:
Oh right I forgot about this. Then the original code looks the best to me :/